test/drafts/ERC20Migrator.test.js

```diff
@@ -0,0 +1,142 @@
+const { assertRevert } = require('../helpers/assertRevert');
+
+const ERC20Mock = artifacts.require('ERC20Mock');
+const ERC20Mintable = artifacts.require('ERC20Mintable');
+const ERC20Migrator = artifacts.require('ERC20Migrator');
+
+const BigNumber = web3.eth.BigNumber;
+
+require('chai')
+  .use(require('chai-bignumber')(BigNumber))
+  .should();
+
+contract('ERC20Migrator', function ([_, owner, recipient, anotherAccount]) {
+  const ZERO_ADDRESS = '0x0000000000000000000000000000000000000000';
+
+  const totalSupply = 200;
+
+  beforeEach('deploying tokens', async function () {
+    this.legacyToken = await ERC20Mock.new(owner, totalSupply);
+    this.migrator = await ERC20Migrator.new(this.legacyToken.address);
+    this.newToken = await ERC20Mintable.new();
+  });
+
+  describe('beginMigration', function () {
+    it('rejects zero address for token', async function () {
+      await assertRevert(this.migrator.beginMigration(ZERO_ADDRESS));
+    });
+
+    it('reverts if not a minter of the token', async function () {
+      await assertRevert(this.migrator.beginMigration(this.newToken.address));
+    });
+
+    it('succeeds if it is a minter of the token', async function () {
+      await this.newToken.addMinter(this.migrator.address);
+      await this.migrator.beginMigration(this.newToken.address);
+    });
+
+    it('reverts the second time it is called', async function () {
+      await this.newToken.addMinter(this.migrator.address);
+      await this.migrator.beginMigration(this.newToken.address);
+      await assertRevert(this.migrator.beginMigration(this.newToken.address));
+    });
+  });
+
+  describe('once migration began', function () {
+    beforeEach('', async function () {
+      await this.newToken.addMinter(this.migrator.address);
+      await this.migrator.beginMigration(this.newToken.address);
+    });
+
+    describe('migrateAll', function () {
+      const baseAmount = totalSupply;
+
+      describe('when the approved balance is higher or equal to the owned balance', function () {
+        const amount = baseAmount;
+
+        beforeEach('approving the whole balance to the new contract', async function () {
+          await this.legacyToken.approve(this.migrator.address, amount, { from: owner });
+        });
+
+        beforeEach('migrating token', async function () {
```

Comment:
This `describe` has two `beforeEach` blocks, which are not independent: I'm not sure if Mocha specifies in which order they'll be run (or even if both will be run), but we should avoid this situation IMO.